src/qt/buynamespage.cpp

```diff
@@ -72,13 +115,90 @@ void BuyNamesPage::onNameEdited(const QString &name)
     }
 }
 
-void BuyNamesPage::onRegisterNameAction()
+void BuyNamesPage::onAsciiNameEdited(const QString &name)
 {
     if (!walletModel)
         return;
 
-    QString name = ui->registerName->text();
+    try
+    {
+        const QString hexName = ASCIIToHex(name);
+        const QString domainName = ASCIIToDomain(name);
+
+        ui->registerNameHex->setText(hexName);
+        ui->registerNameDomain->setText(domainName);
+
+        refreshAvailableError();
+    }
+    catch(InvalidNameString &e)
+    {
+        ui->statusLabel->setText(tr("Not a valid ASCII entry!"));
+    }
+
+}
+
+void BuyNamesPage::onHexNameEdited(const QString &name)
+{
+
+    if (!walletModel)
+        return;
+
+    try{
+        const QString asciiName = HexToASCII(name);
+        const QString domainName = ASCIIToDomain(asciiName);
+        
+        ui->registerNameAscii->setText(asciiName);
+        ui->registerNameDomain->setText(domainName);
+    
+        refreshAvailableError();
+    }
+    catch(InvalidNameString &e)
+    {
+        ui->statusLabel->setText(tr("Not a valid entry!"));
```

Comment:
Maybe say `hex entry` instead of `entry` for consistency with the ASCII codepath?

Reply:
Looks like you missed this one?